First check in for quality/quantity

JapnaLamba opened the pull request:
This is not correct, but I just want to test it anyway!

JapnaLamba committed:
First check in for quality/quantity

This is not correct, but I just want to test it anyway!

shankari reviewed (changes requested):
Need more comments.
Need to be able to run top to bottom without skipping between cells.

JapnaLamba committed:
Small Change

JapnaLamba committed:
Update

JapnaLamba committed:
Berkeley_Survey_Trip
This is the current notebook that we have for evaluating survey data vs. baseline. It only includes a Berkeley walking trip.

JapnaLamba committed:
This is the revised, no-output notebook.

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+   },
+   "outputs": [],
+   "source": [
+    "iphone_id_3 = [UUID(\"c528bcd2-a88b-3e82-be62-ef4f2396967a\")]\n",
```
Instead of calling them iphone_3, iphone_4, etc
why not put in the sensing regime - e.g. high accuracy, 1sec sampling or something to make it more clear.
In this case, because all 4 phones were sampling the same thing, there isn't much difference...

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+    "new_android_df_3.index = new_android_df_3['ts']\n",
+    "new_android_df_4.index = new_android_df_4['ts']\n",
+    "\n",
+    "df1 = pd.DataFrame(new_iphone_df_3, index=new_iphone_df_3.index,columns=['latitude'])\n",
```
use more descriptive names here

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+    "\n",
+    "lines, labels = axes.get_legend_handles_labels()\n",
+    "print lines, labels\n",
+    "axes.legend(lines, [\"iphone3\", \"iphone4\", \"android3\", \"android4\",\"ground_truth\"], loc='best') "
```
add title, description, units etc

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+   "source": [
+    "fig, axes = plt.subplots(nrows=1, ncols=1)\n",
+    "\n",
+    "df2_2 = pd.DataFrame(new_iphone_df_4, index=new_iphone_df_4.index,columns=['longitude'])\n",
```
descriptive names please

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+    "df4_4 = pd.DataFrame(new_android_df_4, index=new_android_df_4.index,columns=['longitude'])\n",
+    "df1_1 = pd.DataFrame(new_iphone_df_3, index=new_iphone_df_3.index,columns=['longitude'])\n",
+    "\n",
+    "df1_1.plot(ax=axes)\n",
```
how is this different from the previous plot? Provide some description/detail

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+   "outputs": [],
+   "source": [
+    "def get_interp_distance_df(android3_ts, android4_ts, ios3_ts, ios4_ts, ground_truth_df, start_ts, end_ts):\n",
+    "    android3_df = android3_ts.get_data_df(\"background/location\", estt.TimeQuery('metadata.write_ts', start_ts, end_ts))\n",
```
didn't we already get these dataframes somewhere earlier?
why not re-use them?

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+    "    interp_df[\"baseline_distances3\"] = baseline_distances_3\n",
+    "    interp_df[\"baseline_distances4\"] = baseline_distances_4\n",
+    "    \n",
+    "    interp_df[\"ground_distances1\"] = np.abs(interp_df.apply(groundDistance1, axis=1))\n",
```
rename away from `ground`

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+    "    interp_df[\"mean_lat\"] = interp_df.apply(meanLat, axis=1)\n",
+    "    interp_df[\"mean_lng\"] = interp_df.apply(meanLng, axis=1)\n",
+    "    \n",
+    "    interp_df[\"mean_lat1\"] = interp_df.apply(meanLat1, axis=1)\n",
```
more descriptive names.
what is `mean_lat` versus `mean_lat1` versus `mean_lat2`?

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+    "    interp_df[\"mean_lng3\"] = interp_df.apply(meanLng3, axis=1)\n",
+    "    interp_df[\"mean_lng4\"] = interp_df.apply(meanLng4, axis=1)\n",
+    "    \n",
+    "    interp_df[\"baseline_distances1\"] = baseline_distances_1\n",
```
ditto.

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+    "    interp_df[\"ground_distances4\"] = np.abs(interp_df.apply(groundDistance4, axis=1))\n",
+    "    \n",
+    "    interp_df[\"ground_distances\"] = np.abs(interp_df.apply(groundDistance, axis=1))\n",
+    "    return interp_df"
```
do we really want to put everything into `interp_df`? Or should we have separate data frames for the interpolated points and the distances? It is fine if there is a reason why they should be together, but I want to know that reason.

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+   },
+   "outputs": [],
+   "source": [
+    "def calDistance1(row):\n",
```
we should really have a single `calDistance`. This was a hack. We should figure out the right way for it to work.

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+   },
+   "outputs": [],
+   "source": [
+    "def groundDistance(row):\n",
```
rename

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+   },
+   "outputs": [],
+   "source": [
+    "def meanLat1(row):\n",
```
single `meanLat` please

shankari replied:
and descriptive name - what's `meanLat1` versus `meanLat2` versus `meanLat3`?

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+   },
+   "outputs": [],
+   "source": [
+    "fig, axes = plt.subplots(ncols=1, nrows=1, sharex=True, sharey=True, figsize=(20,5))\n",
```
For each of the plots, add a caption. The caption should talk about what the plot is showing, as well as what you see in there - e.g. this plot shows the measured values versus the interpolated values and indicates that linear interpolation is able to adequately model the data (or something like that).

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+   },
+   "outputs": [],
+   "source": [
+    "iphone_df_3 = get_points_from_ts_for_range(iphone_ts_3[0],from_ts,\n",
```
How is this different from the earlier `iphone_df_3`? Maybe name it something else?
It would also be good to describe the transformations - we start with a dataframe, we interpolate it, we add baseeline and ground truth information, etc.

Basically, this sequence of steps is a mini-pipeline.
You want to document your mini-pipeline - e.g. 
read data -> interpolate -> calculate baseline -> extract (lat, lng, timestamp) -> ...

shankari replied:
You can have comment cells (e.g. markdown)

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+   },
+   "outputs": [],
+   "source": [
+    "def meanLng1(row):\n",
```
restructure to be common (`meanLng1`, `meanLng2`) -> (`meanLng`)

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+    "Quan = []\n",
+    "ground_distance_mean = []\n",
+    "ground_distance_sum = []\n",
+    "Threshold = sorted([13,12,11,10, 9, 8, 7, 6, 5, 4, 3])\n",
```
what is this? experimental?

shankari commented on Berkeley_Trip.ipynb:
```diff
@@ -0,0 +1,1513 @@
+   "outputs": [],
+   "source": [
+    "d = {}\n",
+    "distances2 = distances[[\"fmt_time\",\"ts\", \"mean_lat\", \"mean_lng\", \"baseline_distances2\", \n",
```
again, what is `distances` and how is it different from `distances`?

shankari reviewed (changes requested):
I have a bunch of detailed comments, but at the high level, I would like to suggest:
- make the variables meaningful
- come up with an overall structure (a mini-pipeline - e.g. interpolate, determine baseline, ....) and document it right at the beginning so people know what to expect
- then have the rest of the document follow that pipeline
- for each graph, add a caption (description of what is in the graph and what is interesting about it)

Once you are done with that, I'll take another look.
